src/components/CartIcon/CartIcon.component.tsx

```diff
@@ -0,0 +1,19 @@
+import type { FC } from 'react';
+
+import { images } from '@/constants/images.constant.ts';
+
+import styles from './CartIcon.module.css';
+
+interface CartIconProps {
+    counter: number;
+    handleCartIcon?: () => void;
+}
+
+export const CartIcon: FC<CartIconProps> = ({ counter, handleCartIcon }) => (
+    <div>
+        <a className={styles.cartButton} onClick={handleCartIcon ?? (() => null)} href="/#">
```

Comment:
Could you use `<button>` there?
You have the bug after clicking cart icon, `href="/#"` may be a reason of the bug.
When a link with` href="/#"` is clicked, it navigates to the root path (/) and then scrolls to the top. But you haven't configured navigation in your project.
This happens after clicking the cart icon on the product card

![image](https://github.com/jaxin007/fe-react-2024/assets/108006836/ec9a51bc-78ee-42c7-9d98-bc9a36c37fb2)